Vic2ToHoI4/Source/Mappers/Government/GovernmentMapper.cpp

```diff
@@ -0,0 +1,138 @@
+#include "GovernmentMapper.h"
+#include "GovernmentMappingFactory.h"
+#include "Log.h"
+#include "V2World/Countries/Country.h"
+
+
+
+namespace
+{
+
+bool governmentMatches(const Mappers::GovernmentMapping& mapping, const std::string& government)
+{
+	return mapping.vic2Government.empty() || (mapping.vic2Government == government);
+}
+
+
+bool rulingIdeologyMatches(const Mappers::GovernmentMapping& mapping, const std::string& rulingIdeology)
+{
+	return mapping.rulingPartyRequired.empty() || (mapping.rulingPartyRequired == rulingIdeology);
+}
+
+
+bool tagMatches(const Mappers::GovernmentMapping& mapping, const std::string& tag)
+{
+	return mapping.tagRequired.empty() || (mapping.tagRequired == tag);
+}
+
+
+bool ideologyIsValid(const Mappers::GovernmentMapping& mapping,
+	 const std::set<std::string>& majorIdeologies,
+	 const HoI4::Ideologies& ideologies)
+{
+	return majorIdeologies.contains(mapping.hoI4GovernmentIdeology) &&
+			 ideologies.subIdeologyIsValid(mapping.hoI4GovernmentIdeology, mapping.hoI4LeaderIdeology);
+}
+
+} // namespace
+
+
+
+std::string Mappers::GovernmentMapper::getIdeologyForCountry(const std::string& sourceTag,
+	 const std::string& sourceGovernment,
+	 const std::string& Vic2RulingIdeology,
+	 bool debug) const
+{
+	std::string ideology = "neutrality";
+	for (const auto& mapping: governmentMap)
+	{
+		if (governmentMatches(mapping, sourceGovernment) && rulingIdeologyMatches(mapping, Vic2RulingIdeology) &&
```

Comment:
Ehh. You're putting your faith that lines with TAG always precede lines with other qualifiers. That's brave, seeing there's no instructions in GovernmentMappings.txt.

I'd run this by tags first and then again for non-tags as those always take precedence, being special cases.

Dditto for all 4 functions below.

Reply:
Most of the items are optional, tags is just the most obvious. With instructions, it should be more clear now that ordering is important.